npm/vite-dev-server/src/resolveConfig.ts

```diff
@@ -70,49 +102,27 @@ export const createViteDevServerConfig = async (config: ViteDevServerConfig, vit
         ],
       },
       entries: [
-        ...specs.map((s) => relative(root, s.relative)),
-        ...(cypressConfig.supportFile ? [resolve(root, cypressConfig.supportFile)] : []),
+        ...specs.map((s) => path.relative(projectRoot, s.relative)),
+        ...(supportFile ? [path.resolve(projectRoot, supportFile)] : []),
       ].filter((v) => v != null),
     },
     server: {
       fs: {
         allow: [
-          root,
+          projectRoot,
           vitePathNodeModules,
-          cypressConfig.cypressBinaryRoot,
+          cypressBinaryRoot,
         ],
       },
       host: '127.0.0.1',
+      // Disable file watching and HMR when executing tests in `run` mode
+      ...(isTextTerminal
+        ? { watch: { ignored: '**/*' }, hmr: false }
+        : {}),
     },
     plugins: [
       Cypress(config, vite),
-      CypressInspect(config),
```

Comment:
A user can push this plugin in themselves, doesn't really make sense for us to support this, especially since it's use is not documented.